Arrowgene.Ddon.Shared/Model/Enemy.cs

```diff
@@ -40,6 +40,8 @@ public Enemy(Enemy enemy)
             Experience = enemy.Experience;
             DropsTable = enemy.DropsTable;
             NotifyStrongEnemy = enemy.NotifyStrongEnemy;
+            Index = enemy.Index;
```

Comment:
Why did we move these? I think the design from Laith was enemy represents the Capcom data and instanced enemy has all the extra information. Also instance enemy represents a unique enemy used by the instance manager where enemy is more generic.

Reply:
While @pacampbell is right that that was my reasoning, I don't think it's entirely wrong for index and subgroup to be a part of this class, as subgroup is part of the fields in the EnemySpawn JSON and index should be (#463). That said, there's some nuance with the PositionIndex field, as the Day/Night enemies system required us to store the index in the InstancedEnemy class